### ISSUE TITLE
NanoGPT: Scatter/Gather not supported with ttir compiler with N300s (triggered by optax.softmax_cross_entropy)

### ISSUE THREAD

**Issue by ugljesak** [tenstorrent/tt-xla#2583]
### Describe the bug

I'm encountering error when running training step for a my JAX NanoGPT on Tenstorrent hardware using `tt-xla`. The error occurs during the jit compilation of the training step. The traceback indicates the failure happens within `optax.softmax_cross_entropy_with_integer_labels`, specifically during the backward pass (gradient calculation) involving `take_along_axis`. The specific error emitted by the compiler is:
```error: 'ttir.scatter' op Batching currently not supported```
This suggests that 'scatter' (and probably 'gather' with it) have not yet been implemented on tt-mlir wheel.

[logs.txt](https://github.com/user-attachments/files/24177180/logs.txt)

### Steps to reproduce the issue

Run training NanoGPT version written in JAX somewhere where Tenstorrent hardware is accessible.
Specifically, any function which has `optax.softmax_cross_entropy_with_integer_labels()`.
You should see the error in the terminal.
For example, you can run my code (I don't have PR yet, but there is my forked branch https://github.com/ugljesak/tt-blacksmith) and there navigate to `blacksmith/experiments/jax/BOUNTIES/nanogpt` and run `python -m test.train_jax` with ensuring main train for loop uses function `train_step` instead of `train_step_onehot` if not already set. Error will pop in terminal.

### Expected behavior

When I run my script it should go with no interruptions and finish training resulting in success.

### Please complete the following environment information

OS version: Ubuntu 22.04
Python version: 3.11.0rc1
JAX version: 0.7.1
